contracts/axelar-auth-verifier/src/contract.rs

```diff
@@ -238,7 +238,7 @@ impl AxelarAuthVerifier {
 
             let WeightedSigner { weight, .. } = signers.signers.get(signer_index).unwrap();
 
-            total_weight = total_weight.add(&weight);
+            total_weight += weight;
```

Comment:
this might overflow by default. rust has a checked_add method, take a look at that